File: app/src/main/java/com/example/culturaverde/principalconsumidoresFragment.java
package com.example.culturaverde;

import android.content.Context;
import android.net.Uri;
import android.os.Bundle;

import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;


public class principalconsumidoresFragment extends Fragment {

    @Override
    public View  onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState){

        return inflater.inflate(R.layout.fragment_principalconsumidores, container, false);
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);


        Button button2 = view.findViewById(R.id.button2);

        button2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Navigation.findNavController(view).navigate(R.id.action_principalconsumidoresFragment_to_menu_desplegable_cons);
            }
        });
    }


}

File: app/src/main/java/com/example/culturaverde/ui/Principalconsumidores/PrincipalConsumidoresFragment.kt
package com.example.culturaverde.ui.Principalconsumidores

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProviders
import androidx.navigation.findNavController
import androidx.navigation.fragment.findNavController
import com.example.culturaverde.R
import kotlinx.android.synthetic.main.fragment_principalconsumidores.*

class PrincipalConsumidoresFragment : Fragment() {

    private lateinit var principalConsumidoresViewModel: PrincipalConsumidoresViewModel

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        principalConsumidoresViewModel =
            ViewModelProviders.of(this).get(PrincipalConsumidoresViewModel::class.java)
        val root = inflater.inflate(R.layout.fragment_principalconsumidores, container, false)

        return root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)


        imagenCategorias.setOnClickListener { view ->
            view.findNavController().navigate(R.id.action_nav_principalconsumidores_to_nav_categoriasFragment2)
        }

        imagenReservas.setOnClickListener { view ->
            view.findNavController().navigate(R.id.action_nav_principalconsumidores_to_nav_reservasFragment)
        }


    }

}